packages/api/src/resolvers/function_resolvers.ts

```diff
@@ -377,7 +381,14 @@ export const functionResolvers = {
         return undefined
       }
 
-      return findUserFeatures(ctx.claims.uid)
+      let features = await getCachedFeatures(ctx.claims.uid)
+      if (!features) {
+        features = await findUserFeatures(ctx.claims.uid)
+
+        await cacheFeatures(ctx.claims.uid, features)
```

Comment:
Why do we need to `await` for the `caching` to be completed? Can't this happen in the asynchronously ?

Reply:
generally we await on all our backend functions, one reason is it can better handle errors.